### PR TITLE
Try UsePythonVersion to work around py2 removal

### LINKED ISSUE #33693: Windows Formatting job failing because it can't find Python on the path
I spoke with @MattGal offline and it seems like the new images don't contain python2 anymore and that is what is breaking this job. @MattGal is going to give a shot on fixing this issue. 

```
##[error]Parameter 'toolPath' cannot be null or empty.
```

https://dev.azure.com/dnceng/public/_build/results?buildId=562848

cc: @dotnet/runtime-infrastructure

### PR DESCRIPTION
**Pull request by MattGal**
Fixes: #33693 

As @safern noted, runtime uses python scripts that assume python is on the path.  With python2's removal from our machines (EOL), there is no longer a python on the machine's path.

### CODE REVIEW

**Comment by BrennanConroy** [comment]
`##[error]Version spec 3.x for architecture x64 did not match any version in Agent.ToolsDirectory.`

**Comment by MattGal** [comment]
> `##[error]Version spec 3.x for architecture x64 did not match any version in Agent.ToolsDirectory.`

This was, sadly, expected;  the buildpool.* agents don't populate the tools folder, so it was just a hope it might magically just work.   Latest commit should help.

File: eng/pipelines/coreclr/templates/format-job.yml [hunk 1]
```diff
@@ -26,7 +26,11 @@ jobs:
     name: ${{ format('format_{0}{1}_{2}', parameters.osGroup, parameters.osSubgroup, parameters.archType) }}
     displayName: ${{ format('Formatting {0}{1} {2}', parameters.osGroup, parameters.osSubgroup, parameters.archType) }}
     helixType: 'format'
-    pool: ${{ parameters.pool }}
+    - ${{ if eq(parameters.osGroup, 'Windows_NT') }}:
```
Comment (safern) [inline eng/pipelines/coreclr/templates/format-job.yml]: I believe the `-` is not needed here.

File: eng/pipelines/coreclr/templates/format-job.yml [hunk 2]
```diff
@@ -26,7 +26,11 @@ jobs:
...
+      pool: 
+        vmImage: 'windows-2019'
+    - ${{ if ne(parameters.osGroup, 'Windows_NT') }}:
```
Comment (safern) [inline eng/pipelines/coreclr/templates/format-job.yml]: Ditto.

**Review by safern — APPROVED** [review]

**Comment by safern** [comment]
Merging since windows formatting job is green and that is the only thing this PR touches. 

Thanks, @MattGal for taking care of this.

**Pull request merged**